### PR TITLE
chore(summary): Add key vault and fusion sql token support

### PR DESCRIPTION
**Pull request by Jonathanio123**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**

[AB#55619](https://statoil-proview.visualstudio.com/787035c2-8cf2-4d73-a83e-bb0e6d937eec/_workitems/edit/55619)

This adds the keyvault support and azure sql token support. I've tested connecting to the Summary CI database which works.

**Testing:**
- [ ] Can be tested
- [ ] Automatic tests created / updated
- [x] Local tests are passing

**Checklist:**
- [ ] Considered automated tests
- [ ] Considered updating specification / documentation
- [x] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

File: src/Fusion.Summary.Api/ConfigurationExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        var configuration = builder.Configuration;
+        var clientId = configuration["AzureAd:ClientId"];
+        var tenantId = configuration["AzureAd:TenantId"];
+        var clientSecret = configuration["AzureAd:ClientSecret"];
+        var keyVaultUrl = configuration["KEYVAULT_URL"];
+
+        if (!string.IsNullOrEmpty(keyVaultUrl))
+        {
+            Console.WriteLine($"Adding key vault using url: '{keyVaultUrl}', client id '{clientId}' and client secret {(string.IsNullOrEmpty(clientSecret) ? "[empty]" : "*****")}");
```
Comment (aleklundeq) [inline src/Fusion.Summary.Api/ConfigurationExtensions.cs:17]: Is it necessary to do print this?

File: src/Fusion.Summary.Api/Program.cs [hunk 2]
```diff
@@ -60,11 +66,14 @@
...
         opts.ClientId = azureAdClientId ?? throw new InvalidOperationException("Missing AzureAd:ClientId");
         opts.ClientSecret = azureAdClientSecret;
+        opts.CertificateThumbprint = certThumbprint;
     });
 });
 
 builder.Services.AddApplicationInsightsTelemetry();
-builder.Services.AddDbContext<SummaryDbContext>(options => options.UseSqlServer(databaseConnectionString));
+builder.Services.AddSqlDbContext<SummaryDbContext>(databaseConnectionString)
+    .AddSqlTokenProvider<SqlTokenProvider>()
```
Comment (aleklundeq) [inline src/Fusion.Summary.Api/Program.cs:75]: Could we just have used "AddFusionTokenProvider" directly here?

Comment (Jonathanio123) [reply src/Fusion.Summary.Api/Program.cs:75]: Unsure if I understand. Something like this?

```suggestion
builder.Services.AddTransient<ISqlTokenProvider, SqlTokenProvider>();
builder.Services.AddSqlDbContext<SummaryDbContext>(databaseConnectionString)
    .AddAccessTokenSupport();
```

AddAccessTokenSupport adds SqlAccessTokenDbInterceptor which requires ISqlTokenProvider

File: src/Fusion.Summary.Api/ConfigurationExtensions.cs (hunk 1 above)
Comment (Jonathanio123) [reply src/Fusion.Summary.Api/ConfigurationExtensions.cs]: Changed it so that it only logs if url or client secret is missing

**Review by aleklundeq — APPROVED** [review]
Other than changing the env.json it looks good. Approved

**Pull request merged**